lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -138,11 +138,19 @@ def check_plugin_manager_ready(self) -> bool:
         """Checks if the plugin manager is ready to run."""
         return (
             self._charm.opensearch.is_node_up()
-            and len(self._charm._get_nodes(True)) == self._charm.app.planned_units()
+            and len(
+                [
+                    x
+                    for x in self._charm._get_nodes(True)
+                    if x.name.startswith(self._charm.app.name)
```

Comment:
This is risky, in case 2 apps: `my-app` and `my-app-2`
```suggestion
                    if "-".join(x.name.split("-")[:-1]) == self._charm.app.name
```